proto/document/v1/document.proto

```diff
@@ -0,0 +1,126 @@
+syntax = "proto3";
+package nitric.document.v1;
+
+import "google/protobuf/struct.proto";
+
+// protoc plugin options for code generation
+option go_package = "nitric/v1;v1";
+option java_package = "io.nitric.proto.document.v1";
+option java_multiple_files = true;
+option java_outer_classname = "Documents";
+option php_namespace = "Nitric\\Proto\\Document\\V1";
+option csharp_namespace = "Nitric.Proto.Document.v1";
+
+// Service for storage and retrieval of simple JSON keyValue
+service Document {
+  // Get an existing document
+  rpc Get (DocumentGetRequest) returns (DocumentGetResponse);
+
+  // Create a new or overwrite an existing document
+  rpc Set (DocumentSetRequest) returns (DocumentSetResponse);
+
+  // Delete an existing document
+  rpc Delete (DocumentDeleteRequest) returns (DocumentDeleteResponse);
+  
+  // Query the document collection (pagination support)
+  rpc Query (DocumentQueryRequest) returns (DocumentQueryResponse);
+  
+  // Query All the document collection (streaming support)
+  rpc QueryAll (DocumentQueryAllRequest) returns (stream DocumentQueryAllResponse);
```

Comment:
For the streamed results, I'd prefer a stream of individual documents rather than a stream of pages. Otherwise, we end up with a double loop implementation on the SDK side and both the Membrane and the client code will need to hold the entire page in memory at the same time:

```
for page in queryresult:
   for doc in page:
```

I'd prefer

```
for doc in results:
   // do something with the doc
```